src/utils/typecheckers.ts

```diff
@@ -0,0 +1,5 @@
+/**
+ * Intended to typecheck enums, but also checks if the value is any of the arbitrary object values
+ */
+export const isEnum = <T extends Object>(object: T, value: any): value is T[keyof T] =>
```

Comment:
I don't like the naming here, for me it is not clear, what this function does. I suppose something like `isEnumValue` or `isValueInEnum` will be more explicit